inngest/src/main/kotlin/com/inngest/InngestFunctionConfigBuilder.kt

```diff
@@ -113,12 +113,20 @@ class InngestFunctionConfigBuilder {
         key: String? = null,
         scope: ConcurrencyScope? = null,
     ): InngestFunctionConfigBuilder {
+        when (scope) {
+            ConcurrencyScope.ENVIRONMENT -> if (key == null) throw InngestInvalidConfigurationException("Concurrency key required with environment scope")
+            ConcurrencyScope.ACCOUNT -> if (key == null) throw InngestInvalidConfigurationException("Concurrency key required with account scope")
+            ConcurrencyScope.FUNCTION -> {}
+            null -> {}
+        }
+
         val c = Concurrency(limit, key, scope)
-        // TODO - Limit concurrency length to 2
         if (this.concurrency == null) {
             this.concurrency = mutableListOf(c)
+        } else if (this.concurrency!!.size == 2) {
```

Comment:
Now that i think of it Kotlin not inferring the non-null type is because different threads could mutate it to `null` between the `null` check and accessing it right? Similar to your other `stepIdsToSeenCount[id]` being still nullable as well.

Reply:
huh interesting I was wondering why the compiler was realizing it wasn't null but that makes sense. Do you know a way around this? I tried wrapping this with a `synchronized` block and it doesn't seem to help 🤔